lib/blocktree/blocktree.go

```diff
@@ -297,29 +298,51 @@ func (bt *BlockTree) Leaves() []Hash {
 	return la
 }
 
-// HighestCommonAncestor returns the highest block that is a Ancestor to both a and b
-func (bt *BlockTree) HighestCommonAncestor(a, b Hash) (Hash, error) {
-	bt.RLock()
-	defer bt.RUnlock()
-
-	an := bt.getNode(a)
-	if an == nil {
+// LowestCommonAncestor returns the lowest common ancestor between two blocks in the tree.
+func (bt *BlockTree) LowestCommonAncestor(a, b Hash) (Hash, error) {
+	// Get nodes to iterate through
```

Comment:
unneeded comment and also kinda wrong, we just get a single node corresponding to block hash A.. Reading more, maybe you meant getting nodes for A and B, but this is a bit confusing to read, I'd just remove :wink: 

```suggestion
```